### PR TITLE
Add LanguageTool

### PR DESCRIPTION
**Pull request by dngray**
Discussion: https://github.com/privacyguides/privacyguides.org/discussions/800

I think this is actually a promising tool, self-hostable, and also keeps your data local.

I think it has particular usefulness to those writing any amount of text, such as blog pots etc.

Even a NeoVim plugin https://www.rockyourcode.com/write-better-in-neovim-with-languagetool/

### CODE REVIEW

**Review by jonaharagon — CHANGES_REQUESTED** [review]
This should be rebased on main, and I think if we add an additional category to this page we should come up with criteria for it. 

Another tool that could fit a new Language Tools section even though it doesn't do the same thing as this might be LibreTranslate, for example.

**Comment by dngray** [comment]
> Another tool that could fit a new Language Tools section even though it doesn't do the same thing as this might be LibreTranslate, for example.

Yeah I rather like that. In fact I think I might try it and see how it works.

**Comment by jonaharagon** [comment]
@dngray you can try a self-hosted LibreTranslate out at https://lt.neat.computer/ - I use the API from this on my Mastodon instance and it works well so I can personally recommend it :)

**Review by mfwmyfacewhen — CHANGES_REQUESTED** [review]

**Review by mfwmyfacewhen — APPROVED** [review]

**Comment by dngray** [comment]
> I use the API from this on my Mastodon instance and it works well so I can personally recommend it :)

Nice. I had been meaning to deploy it myself, but other things came up.

**Review by jonaharagon — APPROVED** [review]

File: docs/productivity.md [hunk 1]
```diff
@@ -166,6 +166,35 @@ In general, we define office suites as applications which could reasonably act a
...
+
+![LanguageTool logo](assets/img/productivity/languagetool.svg#only-light){ align=right }
+![LanguageTool logo](assets/img/productivity/languagetool-dark.svg#only-dark){ align=right }
+
+**LanguageTool** is a multilingual grammar, style and spell checker that supports more than 20 languages. The software is [self-hostable](https://dev.languagetool.org/http-server), and the extensions do not send your input text to their server.
+
+  LanguageTool works with a variety of office suites: Google Documents, Word, OpenOffice and Libreoffice. It also has integration into Gmail, Outlook, Apple Mail and Thunderbird.
+
+[:octicons-home-16: Homepage](https://languagetool.org){ .md-button .md-button--primary }
+[:octicons-info-16:](https://languagetool.org/dev){ .card-link title=Documentation}
```
Comment (redoomed1) [inline docs/productivity.md:185]:
```suggestion
[:octicons-info-16:](https://languagetooler.freshdesk.com/en/support/solutions){ .card-link title=Documentation}
```

File: docs/productivity.md [hunk 2]
```diff
@@ -166,6 +166,35 @@ In general, we define office suites as applications which could reasonably act a
...
+<div class="admonition recommendation" markdown>
+
+![LanguageTool logo](assets/img/productivity/languagetool.svg#only-light){ align=right }
+![LanguageTool logo](assets/img/productivity/languagetool-dark.svg#only-dark){ align=right }
+
+**LanguageTool** is a multilingual grammar, style and spell checker that supports more than 20 languages. The software is [self-hostable](https://dev.languagetool.org/http-server), and the extensions do not send your input text to their server.
+
+  LanguageTool works with a variety of office suites: Google Documents, Word, OpenOffice and Libreoffice. It also has integration into Gmail, Outlook, Apple Mail and Thunderbird.
+
+[:octicons-home-16: Homepage](https://languagetool.org){ .md-button .md-button--primary }
```
Comment (redoomed1) [inline docs/productivity.md:184]:
```suggestion
[:octicons-home-16: Homepage](https://languagetool.org){ .md-button .md-button--primary }
[:octicons-eye-16:](https://languagetool.org/legal/privacy){ .card-link title="Privacy Policy" }
```

File: docs/productivity.md [hunk 3]
```diff
@@ -166,6 +166,35 @@ In general, we define office suites as applications which could reasonably act a
...
+[:octicons-info-16:](https://languagetool.org/dev){ .card-link title=Documentation}
+[:octicons-code-16:](https://github.com/languagetool-org){ .card-link title="Source Code" }
+
+<details class="downloads" markdown>
+<summary>Downloads</summary>
+
+- [:simple-windows11: Windows](https://languagetool.org/windows-desktop)
+- [:simple-apple: macOS](https://languagetool.org/mac-desktop)
+- [:simple-firefoxbrowser: Firefox](https://addons.mozilla.org/firefox/addon/languagetool/)
```
Comment (redoomed1) [inline docs/productivity.md:193]:
```suggestion
- [:simple-firefoxbrowser: Firefox](https://addons.mozilla.org/firefox/addon/languagetool)
```

**Review by redoomed1 — CHANGES_REQUESTED** [review]

**Comment by redoomed1** [comment]
> I think if we add an additional category to this page we should come up with criteria for it.

Are you still thinking about doing this? I think that the [criteria established for Paste services](https://www.privacyguides.org/en/productivity/#minimum-requirements) don't apply to LanguageTool:

> Must implement "zero-trust" end-to-end encryption.

LanguageTool, on its homepage (under the "Create your free account" heading), claims "Texts are securely stored," but I could not find anything about E2EE.

> Must support password-protected files.

I don't think LanguageTool has this feature.

**Comment by dngray** [comment]
The intention was to put this in it's own language section. This tool can be self hosted, and the content was going to be advising that this could be an alternative to DeepL/Google Translate.

Further testing was required.

**Comment by redoomed1** [comment]
> The intention was to put this in it's own language section. This tool can be self hosted, and the content was going to be advising that this could be an alternative to DeepL/Google Translate.

Ah, I see. Thanks for the clarification. I was just a little confused since the criteria for paste services is placed under the LanguageTool card (and Language services section) in the GH actions preview.

**Comment by dngray** [comment]
> advising that this could be an alternative to DeepL/Google Translate.

My confusion there I meant LibreTranslate. This is a grammar tool but same principle applies, it's self hostable and much more privacy friendly. Jonah kindly tested LibreTranslate over at https://lt.neat.computer

**Comment by jonaharagon** [comment]
There was criteria on the productivity page when this PR was created, it was the same that it is now IIRC.

I don't think this needs a separate page, this is basically just a FOSS Grammarly? It seems only useful for writing documents which falls squarely within productivity IMO.

We _should_ modify the productivity criteria to add info about this though.

I think we should add this, and worry about other language-related tools at some future time. LibreTranslate, while I do think it could be added to the site for sure, is actually an entirely different sort of tool. I didn't intend to block adding this tool while we wait to figure out what we want to do with every other language tool out there.

**Comment by dngray** [comment]
> I don't think this needs a separate page, this is basically just a FOSS Grammarly? It seems only useful for writing documents which falls squarely within productivity IMO.

That's fair I guess a language tools heading under productivity page would be sufficient.

**Pull request merged**